drivers/flash/soc_flash_nrf.c

```diff
@@ -488,3 +523,44 @@ static int write(off_t addr, const void *data, size_t len)
 
 	return write_op(&context);
 }
+
+#ifdef CONFIG_NRF52_ANOMALY_242_WORKAROUND
+/* Disable POFWARN by writing POFCON before a write or erase operation.
+ * Do not attempt to write or erase if EVENTS_POFWARN is already asserted.
+ */
+static bool pofcon_enabled;
+static nrf_power_pof_thr_t pofcon_pof_thr;
+
+static int suspend_pofwarn(void)
+{
+	bool enabled;
+	nrf_power_pof_thr_t pof_thr;
+
+	pof_thr = nrf_power_pofcon_get(NRF_POWER, &enabled);
+
+	if (enabled) {
+		nrf_power_pofcon_set(NRF_POWER, false, NRF_POWER_POFTHR_V27);
+
+		/* This check need to be reworked once POFWARN event will be
+		 * served by zephyr.
+		 */
+		if (nrf_power_event_get_and_clear(NRF_POWER,
+						  NRF_POWER_EVENT_POFWARN)) {
```

Comment:
Why is the event actually cleared here? Maybe you could use `nrf_power_event_check()` instead?